octavia_f5/restclient/as3restclient.py

```diff
@@ -78,6 +78,20 @@ def wrapper(self, *args, **kwargs):
 
     return wrapper
 
+def failover_on_connection_error(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        old_bigip = self.active_bigip
+        # Failover until it works or all BigIPs have been tried
+        while True:
+            try:
+                return func(self, *args, **kwargs)
+            except requests.exceptions.ConnectionError as e:
+                self._failover()
+                if self.active_bigip == old_bigip:
+                    raise e # We've tried all possible BigIPs, so give up
+    return wrapper
+
```

Comment:
I really like the implementation as a wrapper 👍 
Although, I would rather like to see another approach then a while True: loop (e.g. move abort condition into loop-condition)

Reply:
Thanks. Honestly, I don't think it is possible to get rid of the `while True` loop without making the code look bad. You'd have to introduce a variable to track whether `func` was called successfully (the return value of the function can't be used since the wrapper has to also work on functions that return `None`). So it would look something like this:
```
def failover_on_connection_error(func):
    @functools.wraps(func)
    def wrapper(self, *args, **kwargs):
        old_bigip = self.active_bigip
        # Failover until it works or all BigIPs have been tried
        success = False
        while not success:
            try:
                result = func(self, *args, **kwargs)
                success = True
            except requests.exceptions.ConnectionError as e:
                self._failover()
                if self.active_bigip == old_bigip:
                    raise e # We've tried all possible BigIPs, so give up
        return result
    return wrapper
```
but I think that's just bloat.

Reply:
Ah I see, jeah pity python doesn't have do-while loops.